### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by josepabellana**

### CODE REVIEW

File: server/index.ts [hunk 1]
```diff
@@ -1,8 +1,38 @@
...
+const { router } = require('./router');
+const cors = require('cors');
+import {auth} from 'express-openid-connect'
+require("dotenv").config();
+
+
+const config = {
+  authRequired: false,
+  auth0Logout:true,
+  secret: process.env.AUTH0_SECRET,
```
Comment (IP1llar) [inline server/index.ts]: Can we have fallback values here?

File: server/router.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// req.isAuthenticated is provided from the auth router
+router.get('/', (req, res) => {
+    res.send(req.oidc.isAuthenticated() ? 'Logged in' : 'Logged out');
+  });
+
+
+//protected route
+router.get('/profile', requiresAuth(), users.getUserData);
+router.post('updateUsername',requiresAuth(), users.updateUsername);
+router.post('updateLearning',requiresAuth(), users.updateUsername)
```
Comment (IP1llar) [inline server/router.ts]: change updateUsername to updateLearning

File: server/tsconfig.json [hunk 3]
```diff
@@ -8,11 +8,10 @@
     "forceConsistentCasingInFileNames": true,            /* Ensure that casing is correct in imports. */
     "strict": true,                                      /* Enable all strict type-checking options. */
     "skipLibCheck": true,
-  "noImplicitAny": true,
+  "noImplicitAny": false,
```
Comment (IP1llar) [inline server/tsconfig.json:11]: No

File: server/router.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+//update game
+//router.post('/updateGame',requiresAuth(), users.updateGameStats)
+
```
Comment (IP1llar) [inline server/router.ts:20]: Maybe split into userRouter and gameRouter when this is implemented

File: server/package.json [hunk 5]
```diff
@@ -4,15 +4,26 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "nodemon ./build/index",
+    "build": "tsc --watch"
```
Comment (IP1llar) [inline server/package.json:8]: Nice

File: server/controllers/user.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        console.log(error);
+        res.status(500).json({ "statusCode": 200, "message": error });
+    }
+}
+
+async function updateUsername(req: express.Request, res: express.Response) {
+    try {
+        const username = req.body.username;
+        const { email } = req.oidc.user;
+        const user = await User.findOne({ email: email })
```
Comment (IP1llar) [inline server/controllers/user.ts:30]: Could use User.updateOne? But this is fine

File: server/index.ts [hunk 7]
```diff
@@ -1,8 +1,38 @@
...
 const app = Express();
+// auth router attaches /login, /logout, and /callback routes to the baseURL
+app.use(auth(config))
+app.use(Express.json());
+app.use(router);
+app.use(cors(corsConfig));
 
-app.use((req, res) => {
-  res.send('Hello World!')
+app.get('/authorized', function (req, res) {
```
Comment (IP1llar) [inline server/index.ts]: Isn't this unsecured?

File: server/models/index.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+require('dotenv').config();
```
Comment (IP1llar) [inline server/models/index.ts]: Shouldn't need this as it's already been called in the root index.ts

File: server/models/index.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+const mongoose = require('mongoose');
+
+mongoose.connect(process.env.DB_URL + process.env.DB_PORT + '/' + process.env.DB_NAME, {
+  useNewUrlParser: true,
```
Comment (IP1llar) [inline server/models/index.ts]: Check if we need these/what they actually do

File: server/models/user.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
+require("dotenv").config();
```
Comment (IP1llar) [inline server/models/user.ts]: Shouldn't need this

**Review by IP1llar — CHANGES_REQUESTED** [review]

File: server/index.ts (hunk 1 above)
Comment (josepabellana) [reply server/index.ts]: The auth0 client and auth0 domain are specific values from the auth0 application account

Comment (josepabellana) [reply server/index.ts]: Added fallbacks on secret and url

File: server/index.ts [hunk 11]
```diff
@@ -8,8 +8,8 @@ require("dotenv").config();
 const config = {
   authRequired: false,
   auth0Logout:true,
-  secret: process.env.AUTH0_SECRET,
-  baseURL: process.env.URL,
+  secret: process.env.AUTH0_SECRET || 'averydifficultpassword',
```
Comment (IP1llar) [inline server/index.ts:11]: Nice

File: server/models/index.ts [hunk 12]
```diff
@@ -1,9 +1,7 @@
-require('dotenv').config();
 const mongoose = require('mongoose');
 
 mongoose.connect(process.env.DB_URL + process.env.DB_PORT + '/' + process.env.DB_NAME, {
-  useNewUrlParser: true,
-  useUnifiedTopology: true,
+  useNewUrlParser: true, //to prevent a warning - will be deprecated soon
```
Comment (IP1llar) [inline server/models/index.ts:4]: Good to know

**Review by IP1llar — APPROVED** [review]

**Pull request merged**